libgalois/src/PropertyGraph.cpp

```diff
@@ -1363,6 +1363,24 @@ katana::EdgeTypeAwareTopology::AdjIndexArray
 katana::EdgeTypeAwareTopology::CreatePerEdgeTypeAdjacencyIndex(
     const PropertyGraph* pg, const EdgeTypeIndex* edge_type_index,
     const EdgeShuffleTopology& topo) noexcept {
+  if (topo.num_nodes() == 0) {
+    KATANA_LOG_VASSERT(
+        topo.num_edges() == 0, "Found graph with edges but no nodes");
+    return AdjIndexArray{};
+  }
+
+  if (edge_type_index->num_unique_types() == 0) {
+    KATANA_LOG_VASSERT(
+        topo.num_edges() == 0, "Found graph with edges but no edge types");
```

Comment:
Not for this PR, but if this can ever happen for real graphs (maybe because all edges have the Unknown type), then this needs to be a `Result`. That said, I think this might not actually be possible for any user input since the Unknown type would always exist and be a unique type.

Reply:
Yes, that shouldn't happen, because even if input has no edge types we assign `kUnknownType` to all edges. That said, I'll review the `EdgeTypeIndex` creation to see if we should put checks there as well (probably should). 

Thanks for the quick review. I messed up a bit by being too confident and merging open-repo PR. Should be resolved in a couple of hours.